internal/uvm/scsi/backend.go

```diff
@@ -0,0 +1,256 @@
+package scsi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/gcs"
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	"github.com/Microsoft/hcsshim/internal/hcs/resourcepaths"
+	hcsschema "github.com/Microsoft/hcsshim/internal/hcs/schema2"
+	"github.com/Microsoft/hcsshim/internal/protocol/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/protocol/guestresource"
+)
+
+// The concrete types here (not the Attacher/Mounter/Unplugger interfaces) would be a good option
+// to move out to another package eventually. There is no real reason for them to live in
+// the scsi package, and it could cause cyclical dependencies in the future.
+
+// Attacher provides the low-level operations for attaching a SCSI device to a VM.
+type Attacher interface {
+	attach(ctx context.Context, controller, lun uint, config *AttachConfig) error
+	detach(ctx context.Context, controller, lun uint) error
+}
+
+// Mounter provides the low-level operations for mounting a SCSI device inside the guest OS.
+type Mounter interface {
+	mount(ctx context.Context, controller, lun uint, path string, config *MountConfig) error
+	unmount(ctx context.Context, controller, lun uint, path string, config *MountConfig) error
+}
+
+// Unplugger provides the low-level operations for cleanly removing a SCSI device inside the guest OS.
+type Unplugger interface {
+	unplug(ctx context.Context, controller, lun uint) error
+}
```

Comment:
I wanted to keep operations on the UVM and operations _inside_ the UVM separate from each other, as I expect we will need to make it easier to separate these parts of the stack in the future (e.g. so that we can manage containers inside a UVM separately from the UVM management itself).

Reply:
isnt that a moot point since we currently dont need that separation and the windows gcs attacher does it automatically?

Reply:
I think VM/container decoupling is the direction we will be moving in, so we can think of this as a preliminary first step towards that goal.